internal/service/instance/nginx_instance_service.go

```diff
@@ -271,3 +278,36 @@ func isFlag(vals []string) bool {
 func isKeyValueFlag(vals []string) bool {
 	return len(vals) == 2
 }
+
+func getLoadableModules(nginxInfo *Info) (modules []string) {
+	var err error
+	if nginxInfo.ConfigureArgs["modules-path"] != nil {
+		modulePath, ok := nginxInfo.ConfigureArgs["modules-path"].(string)
+		if !ok {
+			slog.Warn("error parsing modules-path")
+			return modules
+		}
+		modules, err = readDirectory(modulePath, ".so")
+		if err != nil {
+			slog.Warn("error reading module dir", "dir", modulePath, "error", err)
+			return modules
+		}
+
+		return modules
+	}
+
+	return modules
+}
+
+func readDirectory(dir, ext string) (files []string, err error) {
```

Comment:
Perhaps add a comment explaining what the `ext` is supposed to do, as it's not obvious based on the function's name.